Try what I would be doing at work

rtau opened the pull request.

rtau-cca committed:
Ensure the image built is the same in the future

By using a particular version of the image rather than release tag

rtau-cca committed:
Reduce Docker image size

Use Alpine variant of the image, which is much smaller than default one,
which is based on Debian.

rtau-cca committed:
Use `yarn` instead of `npm`

As yarn.lock is committed, I guess the modules installation are done
using `yarn` rather than `npm`. The same tools should be used during
Docker build.

rtau-cca committed:
Make sure modules version change is notified

By make yarn fail if module version update is needed to complete install

rtau-cca committed:
Prevent unversioned file from interfering build context

By making Docker build ignore the same set of files ignored in GIT

gaplo917 commented:
@rtau 

the `.dockerignore` is not practical when doing developments in local machine.

Also think about, if you have many image/video assets, 1 line source code change will invalidate the docker layer cache and copy all assets and source code again.

rtau-cca committed:
feat: Reduce the files copied on code change

By separating the copy of public directory, which should contains static
contents, and pages, which should contains pages developing into two
separate steps, with copying the less frequently changed public
directory first.

rtau commented:
> @rtau
> 
> the `.dockerignore` is not practical when doing developments in local machine.
> 
> Also think about, if you have many image/video assets, 1 line source code change will invalidate the docker layer cache and copy all assets and source code again.

Change pushed, separate the copying of `public` and `pages` directory into 2 steps, copying `public` directory first, which should contains the less frequently changes assets.

gaplo917 commented:
@rtau  Solution is released, Check it out
